src/lib/Embed.jsx

```diff
@@ -17,6 +17,12 @@ const validateConfig = config => {
   }
 };
 
+const validateCreateReportConfig = config => {
```

Comment:
Let's have 3 functions `validateConfig`, `validateTypeConfig`, `validateModeConfig`. Let `validateConfig` call both based on the config object. Put these function in `utils` and import `validateConfig` and use in this file.

Reply:
done